### PR TITLE
List applying policies in KafkaChannel

### LINKED ISSUE #4038: [control-plane] List applying policies in KafkaChannels `.status.policies`
As the [Eventing Authorization](https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?resourcekey=0-gCE8brkIopDqc4jE7c9_tQ#heading=h.9zw0wrg4io53) feature track describes, target resources of `EventPolicies`, will reflect in their status, which `EventPolicy` is applied to them. 

For example:

```yaml
apiVersion: kafkachannels.messaging.knative.dev/v1
kind: KafkaChannel
metadata:
  name: my-channel
spec:
  ...
status:
  ...
  policies:
    - name: event-policy
      apiVersion: v1alpha1
    - name: another-event-policy
      apiVersion: v1alpha1

  conditions:
    - type: Ready
      status: "True"
    - type: EventPoliciesReady
      status: "True"
```

As the KafkaChannel should support `EventPolicies`, wee need to do the following:

* Update the KafkaChannel reconciler to show applying policies in the KafkaChannels `.status.policies` and set the KafkaChannels `EventPoliciesReady` condition. Also watch EventPolicies for changes.
* Set `EventPoliciesReady` condition to true, in case no policy applies, and set `message` according to the `default-authorization-mode` (e.g. `Default authorization mode is 'allow-all'`). Check the [Default Authorization Mode section](https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?resourcekey=0-gCE8brkIopDqc4jE7c9_tQ#heading=h.c6984bgldrz) in the feature track for additional information.

**Hint:**
* Similar to https://github.com/knative/eventing/issues/7977
* Requires #4069 

**Additional hints for new contributors before starting with this issue:**

1. When the issue has the `Draft` status, the issue is subject to change and thus should not be started to be worked on
2. Make sure you've read and understood the [CONTRIBUTING.md](https://github.com/knative-extensions/eventing-kafka-broker/blob/main/CONTRIBUTING.md) guidelines
3. Make sure you're able to [run Knative Eventing locally](https://github.com/knative/eventing/blob/main/DEVELOPMENT.md#create-a-cluster-and-a-repo) with [Knative Eventing Kafka](https://github.com/knative-extensions/eventing-kafka-broker/blob/main/DEVELOPMENT.md) and [run at least the unit tests](https://github.com/knative-extensions/eventing-kafka-broker/blob/main/DEVELOPMENT.md#run-unit-tests). 
4. Feel free to raise any questions you have either directly here in the issue, in the [#knative-eventing Slack channel](https://cloud-native.slack.com/archives/C04LMU33V1S) or join the [Eventing Workgroup Meeting / Office Hours](https://github.com/knative/community/blob/main/working-groups/WORKING-GROUPS.md#eventing)
5. When you feel comfortable with this issue, feel free to assign it to you (e.g. by commenting `/assign`). Please be aware that we might unassign you, if we don't see any progress from your side to give other contributors also a chance to work on this issue.

### PR DESCRIPTION
**Pull request by rahulii**
Fixes #4038 

## Proposed Changes

- 🎁 List applying EventPolicies in KafkaChannel

**Release Note**

```release-note
List applying EventPolicies in KafkaChannel
```

**Docs**

### CODE REVIEW

File: control-plane/pkg/contract/contract.pb.go
Comment (creydr) [inline control-plane/pkg/contract/contract.pb.go:1]: I think this change is not needed.

**Review by creydr — CHANGES_REQUESTED** [review]
/ok-to-test

File: control-plane/pkg/contract/contract.pb.go
Comment (rahulii) [reply control-plane/pkg/contract/contract.pb.go:1]: uh ho - linters did it, let me revert!

File: control-plane/pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go [hunk 1]
```diff
@@ -51,6 +51,9 @@ const (
 	// KafkaChannelConditionChannelServiceReady has status True when the K8S Service representing the channel
 	// is ready. Because this uses ExternalName, there are no endpoints to check.
 	KafkaChannelConditionChannelServiceReady apis.ConditionType = "ChannelServiceReady"
+
+	ConditionAddressable        apis.ConditionType = "Addressable"
```
Comment (creydr) [inline control-plane/pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go:55]: I guess this is not needed

**Comment by creydr** [comment]
Wondering, why we don't see the GH action runs? :thinking: 

/retest

**Review by creydr — APPROVED** [review]
Thanks a lot for fixing this @rahulii 

/lgtm

**Pull request merged**